threads-csharp/IntroductionToSemaphoreInCsharp/Tests/UnitTests.cs

```diff
@@ -0,0 +1,41 @@
+using IntroductionToSemaphoreInCsharp;
+using System.Text;
+
+namespace Tests;
+
+[TestClass]
+public class UnitTests
+{
+    private StringBuilder ConsoleOutput { get; } = new StringBuilder();
+    const int SleepDelay = 50;
+
+    [TestInitialize]
+    public void Init()
+    {
+        Console.SetOut(new StringWriter(this.ConsoleOutput));
+        this.ConsoleOutput.Clear();
+    }
```

Comment:
I don't know if you've done this before, but we shouldn't rely on Console.SetOut to do the tests. This is an easy way to do any test out there in a console application, but we should avoid it because it's unpredictable and it's not really a test. If we're testing threads or concurrency we should really test that, and not the console output. Do some research on this and try to rewrite these to not include Console.SetOut please.

Reply:
I 've not done this again. I used it as a convenient trade-off wanting to satisfy 2 requirements at the same time and without adding complexity in code:
1. Have a Console output to show in the article what will the code print when we run program.cs
2. Run all code and check that all threads have completed in the Unit Test.

I refactored to how I would do it in normal software:
1. Introduced a ConcurrentQueue to hold all strings generated by the threads.
2. When code completes, I loop/check the output of the ConcurrentQueue.

Can you please review this one and let me know what you think/do you have any improvements to suggest?

One we are happy with the code, I will then update the Article code snippets and code description accordingly.